### PR TITLE
Add basic coding conventions and adjust README

### PR DESCRIPTION
**Pull request by palmskog**

### CODE REVIEW

File: CODING_CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Class TotalOrder {A} (R : relation A) : Prop := {
+  total_order_partial :> PartialOrder R;
+  total_order_trichotomy :> Trichotomy (strict R)
+}.
+```
+
+### Type class instances
+
+- C-style names
+- brackets on separate line indented by two spaces
```
Comment (bmmoore) [inline CODING_CONVENTIONS.md:36]: Why is the indentation and space before semicolons all different from the class declaration above? Also, the example should include an instance with explicit fields.

**Review by bmmoore — CHANGES_REQUESTED** [review]
Readme changes are good.

The coding convention is inconsistent about indentation, and should probably cover record definitions as well.

File: CODING_CONVENTIONS.md (hunk 1 above)
Comment (palmskog) [reply CODING_CONVENTIONS.md]: This was a leftover from somewhere else, I revised and gave an example.

**Comment by palmskog** [comment]
@bmmoore all your comments should be addressed now.

**Review by bmmoore — APPROVED** [review]
Looks good.

I didn't notice the redundancy between meta.yml and the README last time. Is there any way to avoid that?

**Comment by palmskog** [comment]
README.md contains a bunch of boilerplate (shields, lists of dependencies, etc.) that I currently generate from a [Mustache](https://mustache.github.io/) template, namely, [this one](https://github.com/coq-community/templates/blob/070e7d6760d724b565d518dec2df27be9e4a0ff5/README.md.mustache). The tradeoff is that there is redundancy between the metadata file and the various files it generates (which also includes our repo opam file). But I think this is a reasonable tradeoff.

**Pull request merged**